### ISSUE TITLE
Wrong numbers in `@benchmark` histogram

### ISSUE THREAD

**Issue by giordano** [JuliaCI/BenchmarkTools.jl#235]
```
julia> f() = 42
f (generic function with 1 method)

julia> @benchmark f()
BechmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  0.012 ns … 0.038 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     0.013 ns             ┊ GC (median):    0.00%
 Time  (mean ± σ):   0.017 ns ± 0.005 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  ▃   █                                               ▅      
  █▁▁▁█▁▁▁▁▃▁▁▁▁▂▁▁▁▂▁▁▁▁▂▁▁▁▁▂▁▁▁▁▂▁▁▁▂▁▁▁▁▁▁▁▁▁▄▁▁▁▁█▁▁▁▆ ▂
  012 ns         Histogram: frequency by time        024 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.

julia> extrema(ans.times)
(0.012, 0.038)
```
The x-axis of the histogram shows 012 ns and 024 ns, when they should be 0.012 and 0.024 (or 0.038?  where does that 24 come from?).

I'll mention I've seen other cases with weird labels on the x-axis (I think the upper bound was lower than the lower bound), but couldn't get a simple reproducer straight away and forgot to report it.

**Comment by brenhinkeller** [comment]
Here's a similar case: bottom right x-axis label should be 2.40 instead of 240
```
julia> @benchmark foo_lv(999)
BechmarkTools.Trial: 10000 samples with 30 evaluations.
 Range (min … max):  957.033 ns …   4.321 μs  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     989.733 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.089 μs ± 228.984 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  █▇▂▄ ▆  ▄▄     ▅  ▁▁▁  ▂▅                                   ▂ ▂
  ███████▆██▆▅▅▄▅██▆████▇██▆▆▆▆▆▆▆▆▆▆▅▅▅█▇▄▆▅▅▃▅▄▅▅▅▅▁▅▆▃▅▅▄▄▅█ █
  957 ns        Histogram: log(frequency) by time        240 μs <

 Memory estimate: 0 bytes, allocs estimate: 0.
``

**Comment by brenhinkeller** [comment]
I've also seen a case with misplaced decimal point on bottom left x-axis label at least one

**Comment by brenhinkeller** [comment]
cc @tecosaur I suppose in case you have any idea offhand where this is coming from

**Comment by tecosaur** [comment]
I think I know where, and why.
Here's where the time strings are generated
```julia
minhisttime, maxhisttime = remtrailingzeros.(prettytime.(round.(histtimes[[1; end]], sigdigits=3)))
```
the `prettytime` will give something like `"0.024 ns"`.

Then we have `remtrailingzeros`, which uses regex-based string substitution.
However ... I forgot to anchor the regex's and I didn't happen to test a failing case when writing this.

Should be fixed in ed0096a.
